### PR TITLE
Fix profile links not showing for vendor staff

### PR DESCRIPTION
**Pull request by midu-01**
### All Submissions:
* [x] My code follows the [WordPress coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)
* [x] My code satisfies feature requirements
* [x] My code is tested
* [x] My code passes the PHPCS tests
* [x] My code has proper inline documentation
* [ ] I've included related pull request(s) (optional)
* [ ] I've included developer documentation (optional)
* [x] I've added proper labels to this pull request

---

### Changes proposed in this Pull Request:

This PR fixes an issue where **vendor social profile URLs were not visible to vendor staff users** in  
**Vendor Dashboard → Settings → Social Profile**.

Previously, the social profile data was fetched strictly based on the logged-in user ID.  
For staff users, this resulted in fetching staff-level data instead of the parent vendor’s store data, causing empty social profile fields.

This change ensures that:
- When a staff member accesses the Social Profile settings, the **parent vendor ID** is resolved and used
- Social profile URLs saved by the vendor are correctly fetched and displayed for staff users
- Existing vendor behavior remains unchanged

---

### Related Pull Request(s)
* N/A

---

### Closes
* Closes https://github.com/getdokan/dokan-pro/issues/5214#event-21494424531
* Closes https://github.com/getdokan/dokan-pro/issues/5293

---

### How to test the changes in this Pull Request:

1. Log in as a **vendor**
2. Navigate to **Vendor Dashboard → Settings → Social Profile**
3. Add social profile URLs (Facebook, Twitter, Instagram, etc.)
4. Save the settings
5. Log out
6. Log in as a **staff user** associated with the same vendor
7. Navigate to **Settings → Social Profile**
8. Verify that:
   - The previously saved social profile URLs are visible
   - The data matches what the vendor configured
9. Update any social profile URL as staff and save
10. Log back in as vendor and verify the changes persist correctly

---

### Changelog entry

**Fix:** Vendor social profiles not visible for staff users

Fixed an issue where social profile URLs configured by a vendor were not displayed when accessed by vendor staff members. The fix ensures staff users correctly load and manage the parent vendor’s social profile data.

---

### Before Changes

- Staff users saw empty social profile fields
- Social URLs saved by the vendor were not loaded
- REST API requests used the staff user ID instead of vendor store ID

(Screenshots if applicable)

---

### After Changes

- Staff users correctly see vendor social profile URLs
- Social profile data is fetched using the resolved vendor ID
- Vendor and staff experiences are consistent and reliable

(Screenshots if applicable)

---

### Feature Video (optional)

N/A

---

### PR Self Review Checklist:
* [x] Code follows existing style guidelines
* [x] Naming is clear and understandable
* [x] KISS principle followed
* [x] No unnecessary duplication (DRY)
* [x] Readable and maintainable code
* [x] No performance regressions
* [x] No unnecessary complexity
* [x] No grammar or spelling errors

---

### FOR PR REVIEWER ONLY:

> As a reviewer, your feedback should be focused on the idea, not the person. Seek to understand, be respectful, and focus on constructive dialog.  
> As a contributor, I will iterate on this PR based on feedback to ensure the best possible solution.

* [ ] Correct — Does the change fully resolve the issue?
* [ ] Secure — No security risks introduced
* [ ] Readable — Code is easy to understand and maintain
* [ ] Elegant — Solution fits the existing architecture

## Summary by CodeRabbit

* **New Features**
  * Centralized vendor authorization: owner vs. staff resolution with conditional exposure of admin-only fields.
  * Store settings endpoints now reuse shared store authorization and response behavior.

* **Bug Fixes**
  * Clearer 400/403/404 responses for invalid or missing stores.
  * Restricted fields are filtered from updates and views for unauthorized roles; sensitive commission fields hidden when appropriate.

* **Tests**
  * Extensive authorization tests for admin, vendor, vendor‑staff, customer, unauthenticated, and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: includes/REST/StoreController.php [hunk 1]
```diff
@@ -312,15 +312,23 @@ public function get_stores( $request ) {
      */
     public function get_store( $request ) {
         $store_id = (int) $request['id'];
+        $current_user = get_current_user_id();
```
Comment (mrabbani) [inline includes/REST/StoreController.php]: **We may simplify by following the bellow steps:**

- Check the requested id(`$store_id`) is the seller ID using `dokan_is_user_seller( $store_id, true )`. It will ensure that the requested ID is the vendor ID.
-  Otherwise, Fetch the vendor ID for the current user using `dokan_get_current_user_id()`

**Suggested code:**
```php
if ( ! dokan_is_user_seller( $store_id, true ) ) {
   $store_id = dokan_get_current_user_id();
}
//  The existing code
```

File: includes/REST/StoreController.php [hunk 2]
```diff
@@ -361,11 +369,22 @@ public function update_store_permissions_check( $request ) {
...
+        if ( current_user_can( 'dokandar' ) && $current_user === $requested_id ) {
+            return true;
         }
+
+        $staff_vendor_id = get_user_meta( $current_user, '_vendor_id', true );
+        if ( $staff_vendor_id && (int) $staff_vendor_id === $requested_id ) {
+            return true;
+        }
+
+        return false;
```
Comment (mrabbani) [inline includes/REST/StoreController.php]: Do we need to change the permission?

**Review by mrabbani — CHANGES_REQUESTED** [review]

File: includes/REST/StoreController.php (hunk 2 above)
Comment (midu-01) [reply includes/REST/StoreController.php]: Bhai,

Yes, we need to handle the permission here. Staff users should not be able to add or update this profile link unless the appropriate permission is explicitly granted. That is why I implemented this logic specifically for controlling staff update permissions.

Please let me know if you would like any adjustments to this approach.

File: includes/REST/StoreController.php (hunk 1 above)
Comment (midu-01) [reply includes/REST/StoreController.php]: Yes, this can be done using the suggested function. At that time, I did not notice the availability of the dokan_is_user_seller() function. Thank you for pointing this out, bhai. I will update the implementation accordingly.

**Review by mrabbani — CHANGES_REQUESTED** [review]
- Check user capability
- Ensure that the user has the proper access to the requested Store.

**Review by mrabbani — COMMENTED** [review]
## Code Review Summary

Overall, this PR addresses an important issue where vendor staff cannot access social profile links. The approach of using a trait for reusable permission logic is good, and the code structure is clean. However, there are **critical bugs** in the implementation that prevent vendor staff from actually accessing stores, despite the documentation stating they should be able to.

---

## ✅ Positive Aspects

1. **Good use of traits** - The `VendorAuthorizable` trait provides reusable permission logic that can be used across controllers
2. **Clear documentation** - Method docblocks clearly explain the intended behavior
3. **Type hints** - Proper use of PHP type hints improves code safety
4. **Security pattern** - The permission checks follow a good pattern (admin → vendor → staff)

---

## 🔴 Critical Bugs (Must Fix)

### 1. Missing Vendor Staff Check in `check_vendor_authorizable_permission()`

**Location:** `includes/Traits/VendorAuthorizable.php:29-51`

**Issue:** The function documentation states "Vendor staff can access only their assigned vendor store," but the implementation **doesn't actually check for vendor staff**. It only checks if the current user is a seller and if their ID matches the vendor ID, which will always fail for vendor staff.

**Current code:**
```php
if ( dokan_is_user_seller( $current_user_id ) ) {
    return (int) $current_user_id === (int) $vendor_id;
}
return false;
```

**Problem:** `dokan_is_user_seller($current_user_id)` returns `true` for vendor staff (since they have the `dokandar` capability), but the subsequent check `$current_user_id === $vendor_id` will **always fail** for staff members since their user ID is different from the vendor ID.

**Fix required:**
```php
$current_user_id = dokan_get_current_user_id();

// Check if current user is the vendor (exclude staff)
if ( dokan_is_user_seller( $current_user_id, true ) ) { // exclude_staff = true
    return (int) $current_user_id === (int) $vendor_id;
}

// Check if current user is vendor staff with access to this vendor
if ( user_can( $current_user_id, 'vendor_staff' ) ) {
    $staff_vendor_id = (int) get_user_meta( $current_user_id, '_vendor_id', true );
    return $staff_vendor_id > 0 && (int) $staff_vendor_id === (int) $vendor_id;
}

return false;
```

### 2. Missing Requested Store ID Check in `update_store_permissions_check()`

**Location:** `includes/REST/StoreController.php:367-374`

**Issue:** The permission check doesn't verify that the requested store ID matches the current user's vendor ID or their staff's vendor ID.

**Current code:**
```php
return $this->check_vendor_authorizable_permission( $this->get_store_id_for_user() );
```

**Problem:** This only checks if the current user has permission to access their own store, but doesn't verify that the requested store ID (from `$request->get_param('id')`) matches.

**Fix required:**
```php
$requested_id = absint( $request->get_param( 'id' ) );
$user_store_id = $this->get_store_id_for_user();

// For vendor staff, we need to resolve the requested ID to their vendor ID
if ( user_can( get_current_user_id(), 'vendor_staff' ) && $requested_id === get_current_user_id() ) {
    $requested_id = $user_store_id;
}

return $this->check_vendor_authorizable_permission( $requested_id );
```

### 3. Potential Logic Issue in `get_store()` Method

**Location:** `includes/REST/StoreController.php:316`

**Issue:** `get_store_id_for_user()` is called with the requested store ID, but this method is designed to convert a user ID to their vendor ID. If the requested ID is already a vendor ID, this could cause issues.

**Current code:**
```php
$store_id = $this->get_store_id_for_user( (int) $request->get_param( 'id' ) );
```

**Concern:** If the API accepts vendor IDs (store IDs), calling `get_store_id_for_user()` with a vendor ID might incorrectly resolve it if that vendor ID happens to be a user ID that's vendor staff. However, since `dokan_is_user_seller($user_id, true)` excludes staff, if the ID is a vendor, it should return correctly. This might be intentional to allow the API to accept either vendor IDs or user IDs, but the logic could be clearer.

**Recommendation:** Add a comment explaining that the API accepts either a vendor ID or user ID, and if it's a user ID for staff, it resolves to their vendor ID.

---

## ⚠️ Code Quality Issues

### 4. Code Duplication in `update_store()` Method

**Location:** `includes/REST/StoreController.php:389-395`

**Issue:** Manual vendor staff handling duplicates logic that should use `get_store_id_for_user()`.

**Current code:**
```php
if ( user_can( $current_user, 'vendor_staff' ) ) {
    $staff_vendor_id = (int) get_user_meta( $current_user, '_vendor_id', true );
    if ( $staff_vendor_id ) {
        $requested_id = $staff_vendor_id;
    }
}
```

**Recommendation:** Use the trait method instead:
```php
// If current user is staff and requesting their own user ID, resolve to vendor ID
if ( user_can( $current_user, 'vendor_staff' ) && $requested_id === $current_user ) {
    $requested_id = $this->get_store_id_for_user( $current_user );
}
```

---

## 🧪 Testing Recommendations

1. **Test vendor staff can access their vendor's store** via GET endpoint
2. **Test vendor staff can update their vendor's store** via PUT/PATCH endpoint  
3. **Test vendor staff cannot access other vendors' stores**
4. **Test regular vendors cannot access other vendors' stores**
5. **Test admins can access any store**
6. **Test edge cases:** staff without `_vendor_id` meta, invalid vendor IDs, etc.

---

## Summary

The **approach is correct**, but the implementation has **critical bugs** that prevent vendor staff from accessing stores. The main issue is that `check_vendor_authorizable_permission()` doesn't check for vendor staff, even though the documentation says it should. Once these bugs are fixed, the implementation should work as intended.

**Priority fixes:**
1. ✅ **Fix Bug #1** - Add vendor staff check in `check_vendor_authorizable_permission()`
2. ✅ **Fix Bug #2** - Add requested store ID verification in `update_store_permissions_check()`
3. ⚠️ **Refactor Bug #4** - Remove code duplication in `update_store()`

Once these are addressed, this PR should be ready to merge. Great work on identifying the issue and creating a reusable solution!

**Comment by mrabbani** [comment]
## Method Name Suggestions

After reviewing the code, I'd like to suggest better method names that are more readable and follow common PHP naming conventions:

### 1. `check_vendor_authorizable_permission()` → **`can_access_vendor_store()`** ⭐ (Recommended)

**Issues with current name:**
- "authorizable" is awkward/not commonly used
- Too long and verbose
- Not immediately clear what it does

**Why `can_access_vendor_store()` is better:**
- ✅ Clear and readable
- ✅ Follows boolean naming convention (`can_*`)
- ✅ Self-documenting - immediately tells you what it checks
- ✅ Shorter and more concise

**Alternative options:**
- `check_vendor_access()` - Shorter, follows existing `check_*` pattern
- `has_vendor_access()` - Follows `has_*` pattern
- `check_store_access()` - Uses "store" terminology (consistent with existing methods)

### 2. `get_store_id_for_user()` → **`get_vendor_id_for_user()`** ⭐ (Recommended)

**Why this is better:**
- ✅ More specific - "vendor" is clearer than "store" 
- ✅ Better describes what it returns (vendor ID, not just any store ID)
- ✅ More consistent with the codebase terminology

**Alternative options:**
- `resolve_vendor_id()` - Shorter, "resolve" suggests conversion logic
- `get_user_vendor_id()` - Clearer word order

### Recommended Changes

```php
trait VendorAuthorizable {
    
    /**
     * Check if user has vendor permission.
     */
    public function check_permission() {
        return current_user_can( 'dokandar' );
    }

    /**
     * Check whether the current user can access a vendor store.
     * 
     * Admins can access any vendor.
     * Vendors can access only their own store.
     * Vendor staff can access only their assigned vendor store.
     */
    public function can_access_vendor_store( int $vendor_id ): bool {
        // ... implementation
    }

    /**
     * Get vendor ID for a user.
     * 
     * Vendors return their own ID.
     * Vendor staff return their parent vendor ID.
     * Others return 0.
     */
    public function get_vendor_id_for_user( int $user_id = 0 ): int {
        // ... implementation
    }
}
```

**Usage example:**
```php
// Before
if ( ! $this->check_vendor_authorizable_permission( $store_id ) ) {
    // ...
}
$store_id = $this->get_store_id_for_user( $user_id );

// After
if ( ! $this->can_access_vendor_store( $store_id ) ) {
    // ...
}
$vendor_id = $this->get_vendor_id_for_user( $user_id );
```

These names are more readable, shorter, and self-documenting. Consider renaming these methods for better code maintainability! 🚀

**Comment by mrabbani** [comment]
## Testing Guide for Store Setting Controller Changes

### ✅ Automated Tests
All unit tests are passing: **20 tests, 49 assertions**

```bash
vendor/bin/phpunit tests/php/src/REST/StoreSettingControllerTest.php
```

---

### 🔍 Manual Testing Checklist

#### **1. StoreSettingController (v1) Authorization**

**Vendor Access:**
- ✅ Vendor can get own settings: `GET /wp-json/dokan/v1/settings?vendor_id={id}`
- ✅ Vendor can update allowed fields (store_name, phone, address)
- ✅ Vendor cannot update admin commission fields (fields filtered silently)

**Vendor Staff Access:**
- ✅ Vendor staff can get assigned vendor's settings
- ✅ Vendor staff can update allowed fields
- ❌ Vendor staff **CANNOT** update `email` (filtered)
- ❌ Vendor staff **CANNOT** update `password` (filtered)
- ❌ Vendor staff **CANNOT** see admin commission fields

**Admin Access:**
- ✅ Admin can get/update any vendor's settings (including admin fields)

**Customer/Public Access:**
- ✅ Customer can get vendor settings (public data only)
  - Should NOT see: `payment`, `enabled`, `admin_commission*`
  - Should see: `store_name`, `id`, other public fields
- ❌ Customer cannot update vendor settings

#### **2. StoreSettingControllerV2 Endpoints**

Test all 7 endpoints:
1. `GET /wp-json/dokan/v2/settings` - Get settings list
2. `GET /wp-json/dokan/v2/settings/{group_id}` - Get settings group
3. `PUT /wp-json/dokan/v2/settings/{group_id}` - Update settings group
4. `GET /wp-json/dokan/v2/settings/{group_id}/{id}` - Get single setting
5. `PUT /wp-json/dokan/v2/settings/{group_id}/{id}` - Update single setting
6. `GET /wp-json/dokan/v2/settings/{group_id}/{parent_id}/{id}` - Get single field
7. `PUT /wp-json/dokan/v2/settings/{group_id}/{parent_id}/{id}` - Update single field

**Verify field restrictions apply the same way as v1 endpoints**

#### **3. Security Checks**

- ✅ Vendor staff email/password restrictions work
- ✅ Vendor admin commission field restrictions work
- ✅ Authorization checks are in place
- ✅ No sensitive data leakage

#### **4. Edge Cases**

- ✅ Invalid vendor_id returns appropriate error
- ✅ Unauthorized access returns 403/401
- ✅ Cross-vendor access blocked
- ✅ Hook `dokan_rest_store_settings_after_update` fires correctly

---

### 📝 Test User Setup Required

- Admin user
- Vendor user (seller)
- Vendor staff user (assigned to a vendor)
- Customer user

---

### ⚠️ Questions for Reviewers

1. Should vendor staff see admin commission fields? (Currently: NO)
2. Are there additional fields that should be restricted for vendor staff?
3. Should permission callback accept `$request` parameter for consistency?

**Comment by dev-shahed** [comment]
Issue: Existing social URL not retrieving for the role vendor staff.


```
 // Resolve vendor ID: handles both vendor and vendor staff IDs
 $store_id = $this->get_vendor_id_for_user( $requested_id );
 $store = dokan()->vendor->get( $store_id );
```

@midu-01
